modules/core/src/main/scala/com/azavea/stac4s/StacItemAsset.scala

```diff
@@ -1,27 +1,63 @@
 package com.azavea.stac4s
 
 import cats.Eq
+import cats.implicits._
 import io.circe._
+import io.circe.syntax._
 
 final case class StacItemAsset(
     href: String,
     title: Option[String],
     description: Option[String],
-    roles: List[StacAssetRole],
-    _type: Option[StacMediaType]
+    roles: Set[StacAssetRole],
```

Comment:
I know I'm getting _really_ close to yak-shaving here, but I didn't think about this until your comment about `NonEmptyList` in another spot.

This _could_ be [`NonEmptySet`](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/NonEmptySet.scala) - which I didn't know existed until I looked because it's not super prominent in the cats docs

Reply:
I also learned about `NonEmptyVector` and `NonEmptyMap` recently! There are a lot of them it turns out. I'll change it.

Reply:
... I'm now not actually sure it should be `NonEmptySet` -- `roles` [aren't required](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#asset-object), so we'd end up with an `Option[NonEmptySet]` -- we're already providing an empty set if it's not in the json, so I don't know that this changes anything semantically. Also, I think the semantics of an asset with an empty set of roles ("it's some kind of data and I was tired of filling in metadata, please leave me alone") make more sense than querying by an empty list of layer IDs.